### PR TITLE
Cookie handling OKAPI-1104

### PR DESCRIPTION
**Pull request by adamdickmeiss**

### CODE REVIEW

File: okapi-core/src/main/java/org/folio/okapi/util/TokenHeader.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+   * <p>Based on RFC6265.
+   * <a href="https://www.rfc-editor.org/rfc/rfc6265#section-4.2.1">Cookie syntax</a>
+   *    <a href="https://www.rfc-editor.org/rfc/rfc6265#section-4.1.1">Cookie pair syntax</a>
+   * @param cookie cookie string
+   * @param name name to look for
+   * @return value; null if not found
+   * @throws IllegalArgumentException if name occurs multiple times.
+   */
+  static String cookieValue(String cookie, String name) {
+    String[] components = cookie.trim().split("; ");
```
Comment (julianladisch) [inline okapi-core/src/main/java/org/folio/okapi/util/TokenHeader.java:22]: Using existing parser `ServerCookieDecoder.STRICT.decodeAll(cookie)` avoids extensive tests of edge cases.

Comment (adamdickmeiss) [reply okapi-core/src/main/java/org/folio/okapi/util/TokenHeader.java:22]: Ok. Will look into that

File: okapi-common/src/main/java/org/folio/okapi/common/XOkapiHeaders.java [hunk 2]
```diff
@@ -197,4 +197,10 @@ private XOkapiHeaders() {
    * version.
    */
   public static final String OKAPI_MODULE = "okapi";
+
+  /**
+   * Cookie acess token name.
+   */
+  public static final String COOKIE_ACCESS_TOKEN = "accessToken";
```
Comment (julianladisch) [inline okapi-common/src/main/java/org/folio/okapi/common/XOkapiHeaders.java:204]: wiki.example.com might set an accessToken for *.example.com that will be set for okapi.example.com.

To avoid a name clash the cookie name should be more specific, for example folioAccessToken.

Comment (steveellis) [reply okapi-common/src/main/java/org/folio/okapi/common/XOkapiHeaders.java]: I made this change and pushed it here so that we could have all the branches aligned for testing. See what you think: https://github.com/folio-org/okapi/pull/1251/commits/ed437e12699d3bad1c6ba45ed2342e5f6f85f890.

Going to now make this change in the other PRs that are affected.

**Review by steveellis — APPROVED** [review]
